File: testerum-backend/testerum-runner/testerum-runner-api/src/main/kotlin/com/testerum/runner/events/model/TestEndEvent.kt
package com.testerum.runner.events.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum_api.testerum_steps_api.test_context.ExecutionStatus
import com.testerum.model.infrastructure.path.Path
import com.testerum.runner.events.model.position.EventKey
import java.time.LocalDateTime

@Suppress("MemberVisibilityCanBePrivate", "unused")
data class TestEndEvent @JsonCreator constructor(
        @JsonProperty("time")            override val time: LocalDateTime = LocalDateTime.now(),
        @JsonProperty("eventKey")        override val eventKey: EventKey,
        @JsonProperty("testName")        val testName: String,
        @JsonProperty("testFilePath")    val testFilePath: Path,
        @JsonProperty("status")          val status: ExecutionStatus,
        @JsonProperty("durationMillis")  val durationMillis: Long
): RunnerEvent

File: web/web-frontend/src/app/model/test/event/test-end.event.ts
import {Path} from "../../infrastructure/path/path.model";
import {RunnerEvent} from "./runner.event";
import {EventKey} from "./fields/event-key.model";
import {ExecutionStatusEnum} from "./enums/execution-status.enum";
import {RunnerEventTypeEnum} from "./enums/runner-event-type.enum";
import {Serializable} from "../../infrastructure/serializable.model";

export class TestEndEvent implements RunnerEvent, Serializable<TestEndEvent> {
    time: Date;
    eventKey: EventKey;
    eventType: RunnerEventTypeEnum = RunnerEventTypeEnum.TEST_END_EVENT;

    testName: string;
    testFilePath: Path;
    status: ExecutionStatusEnum;
    durationMillis: number;

    deserialize(input: Object): TestEndEvent {
        this.eventKey = new EventKey().deserialize(input["eventKey"]);
        this.time = new Date(input["time"]);
        this.testName = input["testName"];
        this.testFilePath = Path.deserialize(input["testFilePath"]);

        let statusAsString:string = input["status"];
        this.status = ExecutionStatusEnum[statusAsString];

        this.durationMillis = input["durationMillis"];
        return this;
    }

    serialize(): string {
        return undefined;
    }
}
